lib/fileset/internal.nix

```diff
@@ -756,4 +756,27 @@ rec {
     else
       _create fileset._internalBase
         (recurse fileset._internalBase fileset._internalTree);
+
+  # Support for `builtins.fetchGit` with `submodules = true` was introduced in 2.4
+  # https://github.com/NixOS/nix/commit/55cefd41d63368d4286568e2956afd535cb44018
+  _fetchGitSubmodulesMinver = "2.4";
+
+  # Mirrors the contents of a Nix store path relative to a local path as a file set.
+  # Some notes:
+  # - The store path is read at evaluation time.
+  # - The store path must not include files that don't exist in the respective local path.
+  #
+  # Type: Path -> String -> FileSet
+  _mirrorStorePath = localPath: storePath:
```

Comment:
@roberth I'm not sure about that, because this function here (or something like it) would have to be exposed, since it creates the internal representation from the store path. And generally it doesn't seem possible to expose it in a safe way, it's really quiet internal. I'm also not sure if it even has any valid use case other than for `builtins.fetchGit`. We _could_ just ignore that and import the internal function, but I'd rather not violate library boundaries.

The alternative of just having `lib.fileset.gitTracked` for now seems _fine_ to me. The function won't be ambiguous here and it's clear what it does, I don't think it's a problem.